lib/nerves/utils/stream.ex

```diff
@@ -62,6 +79,17 @@ defmodule Nerves.Utils.Stream do
     %{s | timer: Process.send_after(self(), :keep_alive, @timer)}
   end
 
+  defp save_history(s, line) do
+    history = 
+      if length(s.history) >= s.history_lines do
+        [line | history] = s.history
+        history
+      else
+        s.history
+      end
+    %{s | history: history ++ [line]}
+  end
+
```

Comment:
There is almost certainly a better way of doing this. Open to suggestive rewrites

Reply:
Use Erlang's `:queue` module and keep track of the length so that you know when to remove an item for everyone that you insert?